uint8[100] cmd
uint32 MAX_STRLEN = 100
uint32 strlen
uint32 sequence
